source/core/solid_base.cc

```diff
@@ -504,13 +511,15 @@ SolidBase<dim, spacedim>::integrate_velocity(double time_step)
 
           Point<spacedim>     p3 = particle_location + time_step * k3;
           Tensor<1, spacedim> k4;
+          velocity->set_time(initial_time + time_step);
           for (unsigned int comp_i = 0; comp_i < spacedim; ++comp_i)
             k4[comp_i] = velocity->value(p3, comp_i);
 
           particle_location += time_step / 6 * (k1 + 2 * k2 + 2 * k3 + k4);
```

Comment:
again, shouldn't we use `bdf_coeffs` rather than these `1,2,2,1` coefficients? If the scheme here is different than the one used for the fluid resolution, please add a comment to precise the scheme used way ahead, like with the command outside of the loop at l. 489 or just before `k1` initialization

Reply:
It's an implicit RK4 scheme. i'll ad a note regarding this. We advect the particles with an explicit scheme and not an implicit one since anyway the motion of the particle is decoupled from that of the fluid,